### ISSUE TITLE
Calculate Marker/Attribute "Affects Relationship" with mmSolver command Arguments

### ISSUE THREAD

**Issue by david-cattermole** [david-cattermole/mayaMatchMoveSolver#157]
### Feature
Pre-compute an "affects relationship" between Markers and Attributes, using the `maya.cmds.mmSolver` command arguments directly.

The Solver UI and the solver core would benefit from having a simple, non-expensive way of knowing "does Marker A affect Attribute B?".

It is important that we use the same commands argument and the evaluation engine as the `maya.cmds.mmSolver` command, which is only available in the C++ plug-in base. We do not wish to re-implement the same logic in both Python and C++, therefore we should pick the lower-level of the two (C++).

Some of this functionality is available in the Python module `mmSolver.utils.nodeaffects`, however this is a utility function and does not have the logic of `maya.cmds.mmSolver` embedded into it.

Relates to issues #11 and #27.

### Software Versions
- mmSolver version: latest development branch
- Maya version: all
- Operating System (OS): All

**Comment by david-cattermole** [comment]
This feature is now complete and ready to be merged into the develop branch, ready for release in the next version.
This feature is waiting on #27 to be finalised.

The feature uses a `mmSolverAffects` command to calculate the required 'affects' relationship and can be used to query (without modification), or to bake down the affects relationship into attributes stored on the Marker. The attribute names will be named based on the attribute, and can be used to look-up the affects relationship "does Marker X affect Attribute Y?".
